interface/forms/LBF/new.php

```diff
@@ -738,41 +1070,361 @@ function validate(f) {
     }
 }
 
-    end_group();
+// Close all open groups.
+if ($group_table_active) {
+    end_row();
+    echo " </table>\n";
+    $group_table_active = false;
+}
+while (strlen($group_levels)) {
+    $gname = $grparr[$group_levels]['grp_title'];
+    $group_levels = substr($group_levels, 0, -1); // remove last character
+    // No div for an empty group name.
+    if (strlen($gname)) echo "</div>\n";
+}
+
+$display_style = 'none';
+
+if (isset($LBF_SERVICES_SECTION) || isset($LBF_DIAGS_SECTION)) {
+    $fs->loadServiceItems();
+}
+
+if (isset($LBF_SERVICES_SECTION)) {
+
+    // Create the checkbox and div for the Services Section.
+    echo "<br /><span class='bold'><input type='checkbox' name='form_cb_fs_services' value='1' " .
+      "onclick='return divclick(this, \"div_fs_services\");'";
+    if ($display_style == 'block') echo " checked";
+    echo " /><b>" . xlt('Services') . "</b></span>\n";
+    echo "<div id='div_fs_services' class='section' style='display:" . attr($display_style) . ";'>\n";
+    echo "<center>\n";
+    $display_style = 'none';
+
+    // If there are associated codes, generate a checkbox for each one.
+    if ($LBF_SERVICES_SECTION) {
+        echo "<table cellpadding='0' cellspacing='0' width='100%'>\n";
+        $cols = 3;
+        $tdpct = (int) (100 / $cols);
+        $count = 0;
+        $relcodes = explode(';', $LBF_SERVICES_SECTION);
+        foreach ($relcodes as $codestring) {
+            if ($codestring === '') continue;
+            $codes_esc = htmlspecialchars($codestring, ENT_QUOTES);
+            $cbval = $fs->genCodeSelectorValue($codestring);
+            if ($count % $cols == 0) {
+              if ($count) echo " </tr>\n";
+              echo " <tr>\n";
+            }
+            echo "  <td width='$tdpct%'>";
+            echo "<input type='checkbox' id='form_fs_services[$codes_esc]' " .
+              "onclick='fs_service_clicked(this)' value='" . attr($cbval) . "'";
+            if ($fs->code_is_in_fee_sheet) echo " checked";
+            list($codetype, $code) = explode(':', $codestring);
+            $crow = sqlQuery("SELECT code_text FROM codes WHERE " .
+              "code_type = ? AND code = ? AND active = 1 " .
+              "ORDER BY id LIMIT 1",
```

Comment:
What kind of codes is this for. If possibly includes external type codes, then should be using function in custom/code_types.inc.php
(in this case, the lookup_code_descriptions() function)

Reply:
OK. I assume this also works for native (internal) code types?

Reply:
yep, it works for all code types (internal and external) :)